packages/components/src/popover/index.js

```diff
@@ -250,7 +252,39 @@ const Popover = (
 		__unstableShift
 			? shift( {
 					crossAxis: true,
-					limiter: limitShift(),
+					limiter: limitShift( {
+						offset: ( { placement: currentPlacement } ) => {
+							// The following calculations are aimed at allowing the floating
+							// element to shift fully below the reference element, when the
+							// reference element is in a different document (i.e. an iFrame).
+							if ( ownerDocument === document ) {
+								return 0;
+							}
+
+							// The main axis (according to floating UI's docs) is the "x" axis
+							// for 'top' and 'bottom' placements, and the "y" axis for 'left'
+							// and 'right' placements.
+							const mainAxis = isTopBottomPlacement(
+								currentPlacement
+							)
+								? 'x'
+								: 'y';
+							const crossAxis = mainAxis === 'x' ? 'y' : 'x';
+
+							const crossAxisModifier = hasBeforePlacement(
+								currentPlacement
+							)
+								? -1
+								: 1;
+
+							return {
+								mainAxis: -frameOffset[ mainAxis ],
```

Comment:
Why do we always return the negative here?

Reply:
In my tests, setting value to a positive number never didn't lead to the behavior that we were looking for 🤷  Although it would be great if @talldan (who also has experience working and debugging these middlewares) could confirm that this is the correct "math"

Reply:
Instinctively, I would say that the we change sign of the `crossAxis` to take into account the same change of sign that the `offset` middleware applies to its `mainAxis`. Keeping in mind that the two middlewares have opposite definitions for cross and main axis, this means that the change of sign always applied to the axis (always `x` or always `y`)

Reply:
Yes, it seems the offset plays its part here and the 'left` behavior issue is exactly 2 times the iframe offset.. I'm not sure yet if `shift` uses the `offset` value from the middleware and for other calculation takes into account the `iframe` so its doubled..

Having said that, I think if the popover is not vertical it should stick to the limits of the anchored element and not go above or below..

Reply:
Looking at the source code, it looks like the offset from the `offset` middleware doesn't get taken into account for the `mainAxis`:

https://github.com/floating-ui/floating-ui/blob/15d1f49b966f9cf472141749f6ebdb4c75e62d5d/packages/core/src/middleware/shift.ts#L178-L185

while it does get taken into account for the `crossAxis`:

https://github.com/floating-ui/floating-ui/blob/15d1f49b966f9cf472141749f6ebdb4c75e62d5d/packages/core/src/middleware/shift.ts#L197-L206

This discrepancy may be the root cause for the unexpected behavior that we've been experiencing on this PR. I've opened https://github.com/floating-ui/floating-ui/issues/1859, let's see how that goes